File: backend/src/main/java/br/ufpr/sept/so2/modules/iam/api/dto/PrimeiroAcessoRequest.java
package br.ufpr.sept.so2.modules.iam.api.dto;

import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotNull;

public record PrimeiroAcessoRequest(
        @NotBlank(message = "Informe a nova senha") String novaSenha,
        @NotNull(message = "Aceite a política de privacidade") Boolean aceiteTermos
) {
}

File: backend/src/main/kotlin/br/ufpr/sept/so2/modules/iam/api/dto/PrimeiroAcessoRequest.kt
package br.ufpr.sept.so2.modules.iam.api.dto

import jakarta.validation.constraints.NotBlank
import jakarta.validation.constraints.NotNull

data class PrimeiroAcessoRequest(
    @field:NotBlank(message = "Informe a nova senha")
    val novaSenha: String,
    @field:NotNull(message = "Aceite a política de privacidade")
    val aceiteTermos: Boolean?,
)
